src/core_functions/aggregate/README.md

```diff
@@ -0,0 +1,221 @@
+The helpers can vary by the number of arguments, and we will refer to them simply as `info` below.
+Consult the code for details on what is available.
+
+### Size
+
+```cpp
+size()
+```
+
+`State`s are allocated in memory blocks by the various operators
+so each aggregate has to tell the operator how much memory it will require.
+Note that this is just the memory that the aggregate needs to get started -
+it is perfectly legal to allocate variable amounts of memory
+and storing pointers to it in the `State`.
+
+### Initialize
+
+```cpp
+initialize(State *)
+```
+
+Construct a _single_ empty `State` from uninitialized memory.
+
+### Destructor
+
+```cpp
+destructor(Vector &state, AggregateInputData &info, idx_t count)
+```
+
+Destruct a `Vector` of state pointers.
+If you are using a template, the method has the signature
+
+```cpp
+Destroy(State &state, AggregateInputData &info)
+```
+
+### Update and Simple Update
+
+```cpp
+update(Vector inputs[], AggregateInputData &info, idx_t ninputs, Vector &states, idx_t count)
+```
+
+Accumulate the input arguments for each row into the corresponding `State`.
+The template generator methods for `update` operations are (a bit confusingly) called `ScatterUpdate`s
+because the arguments are scattered to the various states.
+
+```cpp
+simple_update(Vector inputs[], AggregateInputData &info, idx_t ninputs, State *state, idx_t count)
+```
+
+Accumulate the input arguments for each row into a single `State`.
+Simple updates are used when there is only one `State` being updated, 
+usually for `SELECT` queries with no `GROUP BY` clause.
+They are defined when an update can be performed more efficiently in a single tight loop.
+There are some other places where this operations will be used if available
+when the caller has only one state to update.
+The template generator methods for simple updates are just called `Update`s.
+
+The template generators use two methods for single rows:
+
+```cpp
+ConstantOperation(State& state, const Arg1Type &arg1, ..., AggregateInputInfo &info, idx_t count)
+```
+
+Called when there is a single value that can be accumulated `count` times.
+
+```cpp
+Operation(State& state, const Arg1Type &arg1, ..., AggregateInputInfo &info)
+```
+
+Called for each tuple of argument values with the `State` to update.
+
+### Combine
+
+```cpp
+combine(Vector &sources, Vector &targets, AggregateInputData &info, idx_t count)
+```
+
+Merges the source states into the corresponding target states.
+If you are using template generators, 
+the generator is `StateCombine` and the method it wraps is:
+
+```cpp
+Combine(const State& source, State &target, AggregateInputData &info)
+```
+
+Note that the `sources` should _not_ be modified for efficiency because the caller may be using them
+for multiple operations(e.g., window segment trees).
+If you wish to combine destructively, you _must_ define a `window` function.
```

Comment:
Nice! Can we add what happens if we leave out the `combine` entirely? Because it is optional, and it might be confusing to understand what happens to the states in that case.

Also, I believe this is the first time you mention segment trees, and I think it might not be very clear/obvious to the reader what these are. For example, 'window segment trees, which are a data structure to store intervals of windows').